src/app.js

```diff
@@ -7,6 +7,19 @@ export function app(app) {
   var node
   var element
 
+  function hydrate(elm) {
+    return elm == null ? undefined : {
+      tag: elm.tagName,
+      data: {},
+      children: [].map.call(elm.childNodes, function(child){hydrate(child)})
+    }
+  }
+
+  if (root.hasChildNodes !== undefined && root.hasChildNodes()) {
+    element = root.children[0]
+    node = hydrate(element)
+  }
+
```

Comment:
I never used SSR (bouh) but I think I understand the general problem: how to differentiate markup generated by SSR from "lambda" markup that lives on its own. One simple solution could be to add an attribute to the root of the SSR rendered DOM element (say `data-ssr` or whatever). So the the client HyperApp have info on how to hydrate it. Tell me if I misunderstood something.